paz/pipelines/pose.py

```diff
@@ -468,3 +472,297 @@ def _build_name_to_sizes(self):
+
+        # pose processors
+        self.match_poses = pr.MatchPoses(model.prior_boxes)
+        self.concatenate_poses = pr.ConcatenatePoses()
+        self.concatenate_scale = pr.ConcatenateScale()
+
+        # pipeline
+        self.add(pr.UnpackDictionary(['image', 'boxes', 'rotation',
+                                      'translation_raw', 'class', 'mask']))
+        self.add(pr.ControlMap(pr.LoadImage(), [0], [0]))
+        self.add(pr.ControlMap(pr.LoadImage(), [5], [5]))
+        if split == pr.TRAIN:
+            self.add(pr.ControlMap(self.augment_colorspace, [0], [0]))
+            self.add(pr.ControlMap(self.augment_6DOF, [0, 1, 2, 3, 5],
+                                   [0, 1, 2, 3, 5]))
+        self.add(pr.ControlMap(self.preprocess_image, [0], [0, 1, 2]))
+        self.add(pr.ControlMap(self.scale_boxes, [3, 1], [3], keep={1: 1}))
+        self.add(pr.ControlMap(self.preprocess_boxes, [4], [5], keep={4: 4}))
+        self.add(pr.ControlMap(pr.TransformRotation(num_pose_dims), [3], [3]))
+        self.add(pr.ControlMap(self.match_poses, [4, 3], [3], keep={4: 4}))
+        self.add(pr.ControlMap(self.match_poses, [4, 5], [7], keep={4: 4}))
+        self.add(pr.ControlMap(self.concatenate_poses, [3, 8], [8],
+                               keep={3: 3}))
+        self.add(pr.ControlMap(self.concatenate_scale, [8, 1], [8]))
+        self.add(pr.SequenceWrapper(
+            {0: {'image': [size, size, 3]}},
+            {4: {'boxes': [len(model.prior_boxes), 4 + num_classes]},
+             7: {'transformation': [len(model.prior_boxes),
+                                    3 * num_pose_dims + 2]}}))
+
+
+class EfficientPosePreprocess(Processor):
+    """Preprocessing pipeline for EfficientPose.
+
+    # Arguments
+        model: Keras model.
+        mean: Tuple, containing mean per channel on ImageNet.
+        camera_matrix:  Array of shape `(3, 3)` camera matrix.
+        translation_scale_norm: Float, factor to change units.
+            EfficientPose internally works with meter and if the
+            dataset unit is mm for example, then this parameter
+            should be set to 1000.
+    """
+    def __init__(self, model, mean=RGB_LINEMOD_MEAN,
+                 camera_matrix=LINEMOD_CAMERA_MATRIX,
+                 translation_scale_norm=1000.0):
+        super(EfficientPosePreprocess, self).__init__()
+
+        self.compute_resizing_shape = pr.ComputeResizingShape(
+            model.input_shape[1])
+        self.preprocess = pr.SequentialProcessor([
+            pr.SubtractMeanImage(mean),
+            pr.PadImage(model.input_shape[1]),
+            pr.CastImage(float),
+            pr.ExpandDims(axis=0)])
+        self.compute_camera_parameter = pr.ComputeCameraParameter(
+            camera_matrix, translation_scale_norm)
+
+    def call(self, image):
+        resizing_shape, image_scale = self.compute_resizing_shape(image)
+        resize_image = pr.ResizeImage(resizing_shape)
```

Comment:
This is not a good idea since one is instantiating and destroying a class at every call. Rather, make a processor that takes as input the resizing shape. So in the constructor build resize_image = pr.ResizeImageWithShape() and then this could be applied in the call as resize_image(image, image_shape)